trashtalk/factories.py

```diff
@@ -53,7 +53,15 @@ def register_all_blueprints(app):
 
 def location_factory(data):
     location = Location(
-        number=data.get('address'),
+        number=data.get('number').strip(),
+        street=data.get('street').strip(),
+        cross_street=data.get('cross_street'),
+        city=data.get('city').strip(),
+        state=data.get('state').strip(),
+        zipcode=data.get('zipcode').strip(),
+        county=data.get('county').strip(),
+        district=data.get('district').strip(),
+        country=data.get('country').strip(),
```

Comment:
These parameters assume that geocoding returns a consistent format that can be reliably parsed. In my experience, the geocode results are too unpredictable to parse this precisely. Consequently, I designated **number** to hold whatever **geocode** returned for *address.*

Reply:
The current and original implementation accounts for the known scenarios: private addresses and public addresses. We will need to adjust or evolve our strategy as we learn of new scenarios.